app/src/views/private/components/export-sidebar-detail/export-sidebar-detail.vue

```diff
@@ -101,6 +101,8 @@ export default defineComponent({
 				if (props.search) {
 					params.search = props.search;
 				}
+			} else {
+				params.limit = -1;
```

Comment:
Instead of making it -1 for everything (and by doing so most likely causing issues on collections of thousands of items), we should add a dropdown in the app to choose between the "regular" numbers (eg 25/50/100/150/250[^1]).

[^1]: Numbers out of the top of my head, we should double check on the tabular layout for what we use there.

Reply:
But this is what the "Use current filters and settings" is used for, right?

Reply:
Yeah, but that doesn't work as expected for most layouts.. I think we want to move to a more wizard like approach where exporting opens a modal for all these settings! That way  we can have one consistent GUI for putting together your export 🙂 

@benhaynes could you do some design / UX exploration around what that modal could look like? (Next week! ⛰️🌊⛰️)

Reply:
Perfect. Let's first discuss what we want to make editable (and how) within this dialog. Some questions:

* Sidebar (current) vs Drawer vs Modal
    * Sidebar is the only non-blocking option, but with limited space
* Options
    * Format — Dropdown of CSV, JSON, or XML (do we want other options here for the delimiter, include header, wrap with quotes, etc?)
    * Fields — Choose Fields, and maybe field order
    * Values — Choose raw value vs relational value vs display plaintext (not sure if we'll support Display Plaintext here, since it's on the App side).
    * Limit — Dropdown of choices, should default to Layout's value
    * Sort — Field and direction, should default to Layout's value
    * Filters — Including Search, should default to Layout's value (optional)

Some Layouts will support some of the above options, but not all. So we need to decide on the **default/required** export config (which will always work)... and then the rest of the optional config can be options within the sidebar/drawer/modal and have defaults that pull from the layout (when available).

**Default Export Options:**
* What limit do we use? Or is this a required dropdown in the sidebar?
* Do we export ALL parent fields by default?
* We should usually have a sort field/order... but maybe not for things like Map. Should this default to PK-ASC?

Reply:
I am not sure if we intend to do this now, but we also have spoken about field mapping and  async export (generated file for later download) 🤔

Reply:
> Sidebar (current) vs Drawer vs Modal

I think we should go modal for this one. It's a more in-depth action on the current _context_ rather than a new context like we normally use drawers for 👍🏻 

> (do we want other options here for the delimiter, include header, wrap with quotes, etc?)

I would love to in the future, but that won't be a part of this current sprint

Reply:
OK, here's the form I'm thinking of. I think this is a bit tall/long for a modal, but can't think of any way to get it smaller... and scrolling within a modal is a bit cumbersome. We should consider using a drawer for this (especially if we add more config here later)... other ideas welcome!

![Data Export Settings](https://user-images.githubusercontent.com/522079/150235546-f6f58256-09c3-41df-9e82-1ce066f07725.png)